### PR TITLE
Add messages to NamedTypeSymbol.Construct exceptions

### PR DESCRIPTION
**Pull request by svick**
Fixes https://github.com/dotnet/roslyn/issues/29098.

### CODE REVIEW

File: src/Compilers/CSharp/Portable/CSharpResources.resx [hunk 1]
```diff
@@ -5339,4 +5339,10 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_OutVariableCannotBeByRef" xml:space="preserve">
     <value>An out variable cannot be declared as a ref local</value>
   </data>
+  <data name="CanCreateConstructedFromConstructed" xml:space="preserve">
+    <value>Can't create constructed generic type from another constructed generic type.</value>
+  </data>
+  <data name="CantCreateConstructedFromNongeneric" xml:space="preserve">
+    <value>Can't create constructed generic type from non-generic type.</value>
```
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/CSharpResources.resx]: while we are not 100% consistent here, the general loc feedback is that contractions should be avoided.  So "cannot create".  This also matches messages like "cannot create static class".

Comment (svick) [reply src/Compilers/CSharp/Portable/CSharpResources.resx]: @CyrusNajmabadi The one message I saw that starts with "Can" does use a contraction:

https://github.com/dotnet/roslyn/blob/ddd03771a64f10cbc3abdff4bcb8ad3438667770/src/Compilers/CSharp/Portable/CSharpResources.resx#L4314-L4316

Contractions are also [recommended by the Microsoft Style Guide](https://docs.microsoft.com/en-us/style-guide/word-choice/use-contractions).

That's just to explain why I wrote it that way initially, I have changed it now.

**Comment by svick** [comment]
@jcouv Sorry, I forgot about those, should be fixed now.

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for a second review of this community PR. Thanks

**Comment by jcouv** [comment]
@svick This is ready to merge. We prefer to squash commits when possible.
Could you do that? Alternatively, are you ok if we do it (via GitHub UI)?

**Comment by svick** [comment]
@jcouv Done.

Though I am curious, isn't just using the GitHub UI better in every way? Why would anyone not be ok with that? (I certainly would be okay with it.)

**Comment by jcouv** [comment]
@svick I wasn't sure what the GitHub UI does about PR authors. I'd have to double-check. I want to make sure the contribution is properly attributed :-)

**Comment by jcouv** [comment]
test windows_debug_unit64_prtest please

**Comment by jcouv** [comment]
Merged. Thanks @svick for the contribution!

**Pull request merged**